File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        from collections import defaultdict
        roman_to_num = defaultdict()
        roman_to_num['I'] = 1
        roman_to_num['V'] = 5
        roman_to_num['X'] = 10
        roman_to_num['L'] = 50
        roman_to_num['C'] = 100
        roman_to_num['D'] = 500
        roman_to_num['M'] = 1000
        ans = 0
        for i in range(len(s)-1):
            if roman_to_num[s[i]] >= roman_to_num[s[i+1]]:
                ans += roman_to_num[s[i]]
            else:
                ans -= roman_to_num[s[i]]
        ans += roman_to_num[s[-1]]
        return ans

File: 0013-roman-to-integer/0013-roman-to-integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        map<char,int> values{{'I', 1}, {'V', 5}, {'X', 10}, {'L', 50}, {'C', 100}, {'D', 500}, {'M', 1000}};
        int ans = 0;
        int i = 0;
        while (i < s.length()) {
            char currChar = s[i];
            int currVal = values[currChar];
            int nextVal = 0;
            if (i+1 < s.length()) {
                char nextChar = s[i+1];
                nextVal = values[nextChar];
            }
            if (currVal < nextVal) {
                ans += nextVal - currVal;
                i += 2;
            } else {
                ans += currVal;
                i += 1;
            }
        } 
        return ans;
    }
};